test/spec/ui/flash.js

```diff
@@ -1,9 +1,16 @@
+import { timeout as d3_timeout } from 'd3-timer';
 import { setTimeout } from 'node:timers/promises';
 
+vi.mock('d3-timer');
+
 describe('iD.uiFlash', function () {
     var context;
 
     beforeEach(function() {
+        // With this mock, a timeout of 0 will be immediately resolved, any
+        // other number will never resolve, to make the tests more deterministic.
+        vi.mocked(d3_timeout).mockImplementation((cb, timeout) => timeout === 0 && cb());
```

Comment:
see code comments. Currently these tests have `duration: 200` and `setTimeout: 225`, this change avoids the need for precise timing

Reply:
yeah, this one is pretty weird. I went with your suggestion, and:
* changed the `.duration(100)` to `0`, because it doesn't matter, we just need ≥1 process tick
* also called `d3.timerFlush()` _before_ `setTimeout`. I don't see any purpose in calling it after `setTimeout`

Seems to pass consistently now[^1]


[^1]: using my (very unscientific) test method of running `npm run test:spec` 30 times with random values of [`--max-old-space-size`](https://nodejs.org/api/cli.html#--max-old-space-sizesize-in-mib) while doing CPU-intensive stuff in the background 😆